Tests/GitHubRepositories.tests.ps1

```diff
@@ -14,7 +14,105 @@ $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
 
 try
 {
-    Describe 'Modifying repositories' {
+    # Define Script-scoped, readonly, hidden variables.
+    @{
+        defaultRepoName = ([Guid]::NewGuid().Guid)
+        defaultRepoDesc = "This is a description."
+        defaultRepoHomePage = "https://www.microsoft.com/"
+        defaultRepoTopic = "microsoft"
+        modifiedRepoDesc = "This is a modified description."
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    Describe 'Getting repositories' {
+
+        Context -Name 'For getting a repository' -Fixture {
+            BeforeAll {
+                $repo = New-GitHubRepository -RepositoryName $defaultRepoName -Description $defaultRepoDesc -AutoInit
+            }
+            AfterAll {
+                Remove-GitHubRepository -Uri "$($repo.svn_url)" -Verbose
+            }
+
+            $newRepo = Get-GitHubRepository -RepositoryName $defaultRepoName
+            It 'Should get repository' {
+                $newRepo | Should Not BeNullOrEmpty
+            }
+
+            It 'Name is correct' {
+                $newRepo.name | Should be $defaultRepoName
+            }
+
+            It 'Description is correct' {
+                $newRepo.description | Should be $defaultRepoDesc
+            }
+        }
+
+        Context -Name 'For getting a from default/repository' -Fixture {
+            BeforeAll {
+                $originalOwnerName = Get-GitHubConfiguration -Name DefaultOwnerName
+                $originalRepositoryName = Get-GitHubConfiguration -Name DefaultRepositoryName
+            }
+            AfterAll {
+                Set-GitHubConfiguration -DefaultOwnerName $originalOwnerName
+                Set-GitHubConfiguration -DefaultRepositoryName $originalRepositoryName
+            }
+
+            $repo = Get-GitHubRepository
+            It 'Should get repository' {
+                $repo | Should Not BeNullOrEmpty
+            }
+
+            It 'Owner is correct' {
+                $repo.owner.login | Should be Get-GitHubConfiguration -Name DefaultOwnerName
```

Comment:
I'm confused.  You never set a new `DefaultOwnerName` above...you only saved the current one during your `BeforeAll`.  As written right now, this will fail because the value returned will be `$null`, and `$repo.owner.login` will be non-null.